### ISSUE TITLE
Crash caused by certain # type: ignore statements

### ISSUE THREAD

**Issue by dargueta** [asottile/dead#48]
It appears that `dead` can't handle code that uses MyPy error codes in `type: ignore` directives.

To reproduce:

```py
def foo() -> int:
    return None  # type: ignore[no-any-return]
```

Run `dead` with no arguments:

```
dargueta@tux ~/blah (master)$ dead           
Traceback (most recent call last):
  File "/Users/dargueta/.pyenv/versions/3.8.6/bin/dead", line 8, in <module>
    sys.exit(main())
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/site-packages/dead.py", line 314, in main
    visitor.visit_comment(lineno, s)
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/site-packages/dead.py", line 195, in visit_comment
    ast_obj = ast.parse(part, f'<{self.filename}:{lineno}: comment>')
  File "/Users/dargueta/.pyenv/versions/3.8.6/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,
  File "<blah.py:2: comment>", line 1
    ignore[no-any-return]
                  ^
SyntaxError: invalid syntax
```

The problem is in the `TYPE_IGNORE_RE` regex. The visit_comment() function ignores these `type: ignore` comments, but the regex doesn't match `type: ignore[xyz]`, only `type: ignore`. I recommend changing the regex to:

```py
TYPE_COMMENT_RE.pattern + r'ignore\s*(\[[\w-]+\]\s*)?(#|$)'
#                                    ^^^^^^^^^^^^^^^^
#                                    insert this
```

**Environment:**

* Python: 3.7.6, 3.8.6
* Operating System: MacOS 11.1 (Big Sur)
* `dead` version: 1.3.0

**Comment by asottile** [comment]
would you be interested in sending a patch?

**Comment by dargueta** [comment]
Yes, I can do that tonight or sometime this weekend. Thanks for the amazingly fast response!

**Comment by dargueta** [comment]
Fixed in #49
